### DISCUSSION TITLE
apache/Caddyfile onlyoffice should use http.request.hostport

### DISCUSSION THREAD

**Discussion by Yonch404** [nextcloud/all-in-one#5061] [Questions] [answered]
### Steps to reproduce
1.Use manual install, use reverse proxy, and enable onlyoffice profile.
2.manually fix some other error, then success to connect onlyoffice in settings view.
3.Try to open a document, then it show the document was banned by Content-Security-Policy.

I found that it isn't the CSP was wrong, when browser requesting /onlyoffice/web-apps/apps/documenteditor/main/index_loader.html, the response header 'Location' lost port, which means use 80 automatically. But I use a non-80-443 port, so it will be banned by CSP.

After Debugging, I found that aio-apache Caddyfile use a wrong field http.request.host, the Caddy document said it didn't include port, which will bother the user to manual install Nextcloud. We should use http.request.hostport.

This line: https://github.com/nextcloud/all-in-one/blob/43a51c2a294c2d922d4f265046a33e2863e13867/Containers/apache/Caddyfile#L43

**Comment by szaimen** [comment, marked as answer]
Edit: I cannot reproduce this. Looks like your reverse proxy is not keeping the headers?

**Comment by Yonch404** [comment]
I'm using apache reverse proxy myself, using 127.0.0.1:7080 to visit in the browser. And it will connect to port 11000 on the backend with aio-apache. Since aio-apache sets X-Forwarded-Host header when forwarding to aio-onlyoffice, this header should have a value of 127.0.0.1:7080. If use http.request.host, it will have a value of 127.0.0.1, resulting in the loss of the port, so that my browser will make subsequent requests to 127.0.0.1:80, then result CSP banned.

So, in theory, one should also use http.request.hostport, which comes with the port. After making my own changes, I now have normal access.
